forge/test/LiquidationBotV2.t.sol

```diff
@@ -0,0 +1,284 @@
+    ) internal returns (address, bytes memory) {
+        string[] memory inputs = new string[](8);
+        inputs[0] = "yarn";
+        inputs[1] = "-s";
+        inputs[2] = "ts-node";
+        inputs[3] = "forge/scripts/get-1inch-swap.ts";
+        inputs[4] = vm.toString(address(liquidator));
+        inputs[5] = vm.toString(fromTokenAddress);
+        inputs[6] = vm.toString(toTokenAddress);
+        inputs[7] = vm.toString(swapAmount);
+
+        string memory responseJson = string(vm.ffi(inputs));
+
+        return (
+            abi.decode(vm.parseJson(responseJson, ".target"), (address)),
+            abi.decode(vm.parseJson(responseJson, ".tx"), (bytes))
+        );
+    }
+
+    function swap(address asset) public {
+        address[] memory liquidatableAccounts;
+
+        (
+            address[] memory assets,
+            uint256[] memory collateralReserves,
+            uint256[] memory collateralReservesInBase
+        ) = liquidator.availableCollateral(liquidatableAccounts);
+
+        uint collateralReserveInBase;
+
+        for (uint8 i = 0; i < assets.length; i++) {
+            if (assets[i] == asset) {
+                collateralReserveInBase = collateralReservesInBase[i];
+            }
+        }
+
+        uint actualSwapAmount = CometInterface(comet).quoteCollateral(
+            asset,
+            collateralReserveInBase
+        );
+
+        (address swapTarget, bytes memory swapTransaction) = get1inchSwap(
+            asset,
+            CometInterface(comet).baseToken(),
+            actualSwapAmount
+        );
+
+        address[] memory swapAssets = new address[](1);
+        swapAssets[0] = asset;
+
+        address[] memory swapTargets = new address[](1);
+        swapTargets[0] = swapTarget;
+
+        bytes[] memory swapTransactions = new bytes[](1);
+        swapTransactions[0] = swapTransaction;
+
+        liquidator.absorbAndArbitrage(
+            liquidatableAccounts,
+            swapAssets,
+            swapTargets,
+            swapTransactions
+        );
+    }
+
+    function initialValues() internal returns (uint, int) {
+        return (
+            ERC20(usdc).balanceOf(liquidator_eoa),
+            CometInterface(comet).getReserves()
+        );
+    }
+
+    function runSwapAssertions(
+        address asset,
+        uint initialRecipientBalance,
+        int initialReserves
+    ) internal {
+        // expect that there is only dust (< 1 unit) left of the asset
+        assertLt(CometInterface(comet).getCollateralReserves(asset), 10 ** ERC20(asset).decimals());
+
+        // expect the balance of the recipient to have increased
+        assertGt(ERC20(usdc).balanceOf(liquidator_eoa), initialRecipientBalance);
+
+        // expect the protocol reserves to have increased
+        assertGt(CometInterface(comet).getReserves(), initialReserves);
+
+        // XXX make sure that you're making > 1% of the value of the swap
+    }
+
+    function testLargeWbtcSwap() public {
```

Comment:
these five `textLarge${asset}Swap` tests are fairly repetitive. originally, i wrote this as a single test that looped over each asset and then performed the same actions/checked the same expecatations.

the advantage of having them as separate tests is that it's easier to get a readable stack trace for failures when they are separate. when you have all 5 in the same test and there is a failure, the stack trace is enormous and filled with a number of successful operations.

still undecided on which is preferable.

Reply:
For now, we could keep them separate for easier debugging. But we can always combine them into one test if maintenance ever becomes a pain in the future.